Storage/semantics.h

```diff
@@ -0,0 +1,141 @@
+/*******************************************************************************
+The MIT License (MIT)
+
+Copyright (c) 2016 Dmitry "Dima" Korolev <dmitry.korolev@gmail.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+#ifndef CURRENT_STORAGE_SEMANTICS_H
+#define CURRENT_STORAGE_SEMANTICS_H
+
+#include "../port.h"
+
+#include "container/sfinae.h"
+
+#include "../Blocks/HTTP/api.h"
+
+namespace current {
+namespace storage {
+
+// Helper empty classes for call dispatching, REST and generic cross-container access.
+namespace semantics {
+
+namespace rest {
+struct RESTWithSingleKey {};
+struct RESTWithPairedKey {};
+}  // namespace current::storage::semantics::rest
+
+namespace primary_key {
+struct Key {};
+struct RowCol {};
+}  // namespace current::storage::semantics::primary_key
```

Comment:
Don't the structures described in `rest` and in `primary_key` actually mean the same?